core/src/main/java/com/linecorp/armeria/common/DefaultHttpHeaders.java

```diff
@@ -186,16 +188,16 @@ public HttpHeaders status(HttpStatus status) {
     @Nullable
     @Override
     public MediaType contentType() {
-        final MediaType contentType = this.contentType;
-        if (contentType != null) {
-            return contentType;
-        }
-
         final String contentTypeString = get(HttpHeaderNames.CONTENT_TYPE);
         if (contentTypeString == null) {
             return null;
         }
 
+        final MediaType contentType = this.contentType;
+        if (contentType != null && Ascii.equalsIgnoreCase(contentType.toString(), contentTypeString.trim())) {
```

Comment:
This is slightly safer but still misses e.g. `application/json;    charset=utf-8` - notice spaces within the content type. The safest way would be to store `parsedContentTypeString` as a field to compare with the raw input instead of normalized, but maybe overkill - I feel as if it'd be find to leave out the `trim` though since it's not a complete normalization anyways.

Reply:
Yeah it might be better just remove the `trim` because it's not a normalization. :-)
 Thank you~!